packages/rolldown/src/treeshake/module-side-effects.ts

```diff
@@ -1,21 +1,27 @@
 import { z } from 'zod'
 
-export const HasModuleSideEffectsSchema = z
-  .function()
-  .args(z.string(), z.boolean())
-  .returns(z.boolean())
-export type HasModuleSideEffects = z.infer<typeof HasModuleSideEffectsSchema>
-
 export type ModuleSideEffectsOption = z.infer<
   typeof ModuleSideEffectsOptionSchema
 >
 
-export const ModuleSideEffectsOptionSchema = z.boolean().or(z.string())
+export const ModuleSideEffectsRuleSchema = z
+  .object({
+    test: z.instanceof(RegExp).optional(),
+    external: z.boolean().optional(),
+    sideEffects: z.boolean(),
+  })
+  .refine((data) => {
```

Comment:
NIT: Would be good if this a compile time error.

Reply:
I am working on that.